lib/stdlib/test/peer_SUITE.erl

```diff
@@ -328,6 +336,59 @@ stop_peer(Config) when is_list(Config) ->
     %% shutdown node
     peer:stop(Peer).
 
+shutdown_halt() ->
+    [{doc, "Test that peer shutdown halt wait until node connection is down"}].
+shutdown_halt(Config) when is_list(Config) ->
+    false = shutdown_test(Config, halt, 500, 500, 1000),
+    ok.
+
+shutdown_halt_timeout() ->
+    [{doc, "Test that peer shutdown halt forcefully takes down connection on timeout"}].
+shutdown_halt_timeout(Config) when is_list(Config) ->
+    false = shutdown_test(Config, {halt, 1000}, 5000, 1000, 1500),
+    ok.
+
+shutdown_stop() ->
+    [{doc, "Test that peer shutdown stop wait until node connection is down"}].
+shutdown_stop(Config) when is_list(Config) ->
+    false = shutdown_test(Config, infinity, 500, 500, 2000),
+    ok.
+
+shutdown_stop_timeout() ->
+    [{doc, "Test that peer shutdown stop forcefully takes down connection on timeout"}].
+shutdown_stop_timeout(Config) when is_list(Config) ->
+    false = shutdown_test(Config, 1000, 5000, 1000, 2500),
+    ok.
+
+shutdown_close() ->
+    [{doc, "Test that peer shutdown close does not wait for dist connection"}].
+shutdown_close(Config) when is_list(Config) ->
+    _ = shutdown_test(Config, close, 5000, 0, 200),
+    ok.
+
+shutdown_test(Config, Shutdown, BlockTime, MinWaitTime, MaxWaitTime) ->
+    Options0 = #{shutdown => Shutdown, args => ["-hidden"]},
+    Options = case proplists:get_value(connection, Config) of
+                  undefined -> Options0;
+                  Conn -> maps:put(connection, Conn, Options0)
+              end,
+    {ok, Peer, Node} = ?CT_PEER(Options),
+    erpc:cast(Node,
+              fun () ->
+                      erts_debug:set_internal_state(available_internal_state, true),
+                      erts_debug:set_internal_state(block, BlockTime+100)
+              end),
+    receive after 100 -> ok end,
+    Start = erlang:monotonic_time(millisecond),
+    peer:stop(Peer),
+    End = erlang:monotonic_time(millisecond),
+    WaitTime = End - Start,
+    Res = lists:member(Node, nodes(connected)),
+    io:format("Connected = ~p~nWaitTime = ~p~n", [Res, WaitTime]),
```

Comment:
To my understanding, `ct:print` (or, `pal`, `log`) is the preferred way to print/log something within a test suite.
This allows to add fine-grained filters (e.g. print only when test case fails, because in most cases developers aren't interested in the output of successful cases).

Reply:
Using `ct:pal` now